heat/core/dndarray.py

```diff
@@ -1399,222 +1401,175 @@ def __getitem__(self, key):
         (1/2) >>> tensor([0.])
         (2/2) >>> tensor([0., 0.])
         """
-
+        # print('start getitem', key)
+        # print()
         l_dtype = self.dtype.torch_type()
-        if isinstance(key, DNDarray) and key.gshape[-1] != len(self.gshape):
-            key = tuple(x.item() for x in key)
+        kgshape_flag = False
+        if isinstance(key, DNDarray) and key.ndim == self.ndim:
```

Comment:
Would you mind to add some high-level comments here as to what each of these branches are doing?